### PR TITLE
[CONS BLOG] RAPIDS cuCIM: Porting skimage Code to GPU

### PR DESCRIPTION
**Pull request by bskinn**
## Text styling

- [ ] The blog is written with [plain language](https://www.plainlanguage.gov/guidelines/) (where relevant).
- [x] If there are headers, they use the proper header tags in order to do so (with only one level-one header).
- [x] All links describe where they link to (for example, check the [Quansight labs website](https://labs.quansight.org/)).
- [x] Any kind of styling that the author uses (for example, bold for emphasis) is consistent throughout the blog.

## Non-text contents

- [ ] All content is represented as text (for example, images need alt text and videos need captions or descriptive transcripts).
- [x] If there are emojis, there are not more than three in a row.
- [x] Don't use [flashing gifs or videos](https://www.w3.org/TR/UNDERSTANDING-WCAG20/seizure-does-not-violate.html).
- [ ] If it were to be read as plain text, the blog still makes sense and no information is missing.

### CODE REVIEW

**Comment by ericdatakelly** [comment]
TODO: Contact RAPIDS and have them update their link to our post.

**Comment by ericdatakelly** [comment]
> TODO: Contact RAPIDS and have them update their link to our post.

Or we could forward to the new url.

**Comment by bskinn** [comment]
The URL shouldn't be changing -- if it is, we need to rename the .md file to match.

**Comment by bskinn** [comment]
Weird, why would the link from their post have ever been incorrect?

**Comment by ericdatakelly** [comment]
> Weird, why would the link from their post have ever been incorrect?

Not sure.  I remember testing those links when they went live.

**Comment by bskinn** [comment]
I'll check Wix to see what URL the post originally went up under...it should live there.

We may need to add a redirect as you suggest, though, @ericdatakelly, since the post has been live at this URL for a few months now.

**Comment by ericdatakelly** [comment]
Besides the url issue, this post is ready.  I went ahead and replaced some images with code snippets and a table because they were easy to do and help with accessibility.  The preview displays them well, so I think this works, but of course we can go back to the images if needed.

**Comment by bskinn** [comment]
> I went ahead and replaced some images with code snippets and a table because they were easy to do and help with accessibility.

Great, I think this is what we want in general, for any of these posts. I can't recall for sure, but it's possible these were replaced with images because we didn't have a good code-block formatting option inside Storyblok and we needed _something_ to migrate.

Thanks!

**Comment by bskinn** [comment]
Yeah, looking at a [Wayback snapshot](https://web.archive.org/web/20220526150150/https://www.quansight.com/blog), the URL should be what's linked from NVIDIA. I'll do the following:

- [ ] Rename this post to match the old URL
- [ ] Add a Vercel redirect from where the post has been living on the new site to the old URL

**Comment by ericdatakelly** [comment]
If the path to this post changed, others might have changed too, and there are probably other sites/blogs that reference them.  We should check the other posts too.

**Comment by bskinn** [comment]
> and a table because they were easy to do and help with accessibility

@ericdatakelly, do you know if it's possible to constrain the table width using only Markdown? This table only has two columns but is rendering the full width of the text column, and it doesn't look that good.

**Comment by ericdatakelly** [comment]
> > and a table because they were easy to do and help with accessibility
> 
> @ericdatakelly, do you know if it's possible to constrain the table width using only Markdown? This table only has two columns but is rendering the full width of the text column, and it doesn't look that good.

I don't.  However, I think the platform is interpreting html and css, so it should be possible to use one of these tricks.  
https://stackoverflow.com/questions/36121672/set-table-column-width-via-markdown

Or are you going for pure markdown to avoid future conflicts?

File: apps/consulting/posts/performance-for-image-processing-with-cucim.md [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+from CuPy’s on disk kernel cache in the case of a new Python session).
+
+The necessary changes typically involve:
+
+1. Use `cupy.asarray` to transfer array inputs to cuCIM functions from the host
+   to the GPU.
+2. Replace functions imported from CPU-based library with GPU equivalents as in
+   the following table:
+
+| CPU Module | GPU Module      |
```
Comment (bskinn) [inline apps/consulting/posts/performance-for-image-processing-with-cucim.md:173]: @noatamir / @gabalafou / @trallard -- do any of you know how we could constrain the max width of this table (say, to 50% of the story column width) via some sort of MDX syntax?

**Comment by bskinn** [comment]
> I think the platform is interpreting html and css

The source is being interpreted as MDX (a JSX/MD hybrid), and I'm still not very good at making it work. Left a comment inline requesting assistance.

File: apps/consulting/posts/performance-for-image-processing-with-cucim.md (hunk 1 above)
Comment (bskinn) [reply apps/consulting/posts/performance-for-image-processing-with-cucim.md]: Never mind, adding the table as raw, Tailwind-styled HTML worked well, once I added `.md` and `.mdx` to the Tailwind content search config.

File: apps/consulting/posts/performance-for-image-processing-with-cucim.md [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+from CuPy’s on disk kernel cache in the case of a new Python session).
+
+The necessary changes typically involve:
+
+1. Use `cupy.asarray` to transfer array inputs to cuCIM functions from the host
+   to the GPU.
+2. Replace functions imported from CPU-based library with GPU equivalents as in
+   the following table:
+
+<div class="flex md:justify-center md:pl-0 justify-left pl-16">
```
Comment (bskinn) [inline apps/consulting/posts/performance-for-image-processing-with-cucim.md:173]: Not the most elegant solution in-source, @ericdatakelly, but I think I got the table looking pretty good. See what you think in the [preview](https://consulting-pgp0li5a8-quansight.vercel.app/post/performance-for-image-processing-with-cucim#adding-gpu-support-to-existing-cpu-based-code).

Comment (ericdatakelly) [reply apps/consulting/posts/performance-for-image-processing-with-cucim.md:173]: It looks great!

**Pull request merged**